src/Filters/LinearKalmanFilter.h

```diff
@@ -0,0 +1,57 @@
+#ifndef LINEAR_KALMAN_FILTER_H
+#define LINEAR_KALMAN_FILTER_H
+
+#include "Filter.h"
+#include "../Math/Matrix.h"
+
+namespace mmfs {
+
+class LinearKalmanFilter : public Filter {
+public:
+
+    int measurementSize;
+    int controlSize;
+    int stateSize;
+
+    // Constructors
+    LinearKalmanFilter(int measurementSize, int controlSize, int stateSize);
+    LinearKalmanFilter(Matrix X, Matrix P, Matrix K);
+    LinearKalmanFilter(Matrix X, Matrix P);
+    virtual ~LinearKalmanFilter() = default;
+
+    // Virtual getter methods for matrices, to be overridden by subclasses
+    virtual void initialize() = 0;
+    virtual Matrix getF(double dt) = 0;
+    virtual Matrix getG(double dt) = 0;
```

Comment:
this feels wrong. you shouldn't need to pass dt? At this level of abstration (linear kf) there is no implementation that assumes that dt is needed to find F and G. a linear kf is just a purely math algorithm

Reply:
tho we will always use a time so its fine for now

Reply:
I mean these are pure virtual functions, so they don't really exist at the Linear kalman filter class level, it's just saying that any subclass will need to implement them. `dt` here, exists as like a variable, as in matrix F for example, wouldn't you need a variable like `t` or Delta `t` to represent the actual dynamics equations. I think also in the original code it required `dt` to be passed in so I just copied that

Reply:
yea for anything we do (that I can think of) we will only need to pass dt so lets keep it. just a linear kalman filter doesn't necessarily need anything to define F or might need many args to create F. But this works for us now and thats all that matters